pkg/hostpath/hostpath.go

```diff
@@ -684,11 +731,13 @@ func (hp *hostPath) deleteHostpathVolume(volID string) error {
 		storeVolMapToDisk()
 	}
 	objcache.UnregisterSecretUpsertCallback(volID)
+	objcache.UnregisterConfigMapUpsertCallback(volID)
 	objcache.UnregisterSecretDeleteCallback(volID)
 	objcache.UnregisterConfigMapDeleteCallback(volID)
-	objcache.UnregisterConfigMapUpsertCallback(volID)
-	objcache.UnregisterShareDeleteCallback(volID)
-	objcache.UnregisterShareUpdateCallback(volID)
+	objcache.UnregisterSharedConfigMapDeleteCallback(volID)
+	objcache.UnregisterSharedSecretDeleteCallback(volID)
+	objcache.UnregsiterSharedSecretsUpdateCallback(volID)
+	objcache.UnregisterSharedConfigMapUpdateCallback(volID)
```

Comment:
I feel like you are missing a couple of `Unregister<thing>`here, you might want to double-check

Reply:
double checked and I think I have all of them ... the ordering could be improved ... I'll include that in my next push